src/output/plugins/visualization/SoundInfoCache.hxx

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The Music Player Daemon Project
+
+#ifndef SOUND_INFO_CACHE_HXX_INCLUDED
+#define SOUND_INFO_CACHE_HXX_INCLUDED
+
+#include "output/Timer.hxx"
+#include "pcm/AudioFormat.hxx"
+#include "thread/Mutex.hxx"
+#include "util/AllocatedArray.hxx"
+
+#include <chrono>
+#include <memory>
+
+namespace Visualization {
+
+/**
+ * \brief Thread-safe cache for recent PCM data
```

Comment:
I don't understand the whole point of this class and why it needs to be thread-safe. I also don't understand the documentation text here.
Why "cap the amount of of memory used"? What does that mean?
Why is it called "cache"?